### PR TITLE
added flag to pass log-location and also logged the console messages into log file

### PR DESCRIPTION
**Pull request by devidask27**
## ISSUE(S):

https://platform9.atlassian.net/browse/FT-262

## SUMMARY

Added new flag to pass location to save logs, and also logged the console messages into logfile

## ISSUE TYPE

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that may cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## RELATED ISSUE(S):

https://platform9.atlassian.net/browse/FT-361

#### Manual

**Flag to pass log location**
Screenshot 2022-03-28 at 12 57 48 PM
**User defined log location to save logs**
Screenshot 2022-03-28 at 12 59 21 PM
**Default log location**
Screenshot 2022-03-28 at 1 02 07 PM

#### Reviewers

/cc @anupbarve @nikhilbandi

### CODE REVIEW

**Comment by devidask27** [comment]
Log file
[pf9ctl-20220328.log](https://github.com/platform9/pf9ctl/files/8361663/pf9ctl-20220328.log)

File: cmd/root.go [hunk 1]
```diff
@@ -82,6 +85,13 @@ func initConfig() {
 		viper.SetConfigName(".pf9ctl")
 	}*/
 
+	if rootCmd.Flags().Changed("log-dir") {
+		util.Pf9Log = filepath.Join(logDirPath, "pf9ctl.log")
+		util.Pf9LogLoc = logDirPath
+	} else {
```
Comment (anupbarve) [inline cmd/root.go:91]: What happens when the logDirPath doesn't exist ?
The directory should be created with appropriate permissions.

File: cmd/root.go [hunk 2]
```diff
@@ -82,6 +85,13 @@ func initConfig() {
...
+		util.Pf9LogLoc = "pf9/log"
```
Comment (anupbarve) [inline cmd/root.go:92]: Add to constants.

File: pkg/util/constants.go [hunk 3]
```diff
@@ -95,7 +95,7 @@ var (
 	OVFLoc    = "/etc/systemd/system/ovf.service"
 	VarDir    = "/var/log/pf9"
 	EtcDir    = "/etc/pf9"
-	Pf9LogLoc = "pf9/log"
+	Pf9LogLoc string
```
Comment (anupbarve) [inline pkg/util/constants.go:98]: why ?

**Review by anupbarve — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 1 above)
Comment (devidask27) [reply cmd/root.go:91]: if the logDirPath doesn't exist then it throughs error with message no such file or directory
Screenshot 2022-04-04 at 10 37 32 AM

File: pkg/util/constants.go (hunk 3 above)
Comment (devidask27) [reply pkg/util/constants.go:98]: Previously default log locations was "pf9/log" but now its depends on user so assigning value based on flag.

File: cmd/root.go (hunk 2 above)
Comment (devidask27) [reply cmd/root.go]: Done

File: cmd/root.go (hunk 1 above)
Comment (anupbarve) [reply cmd/root.go:91]: We need to create the directory via code as mentioned above instead of erroring out.

**Review by anupbarve — APPROVED** [review]

**Pull request merged**